cheddar/cheddar-server/src/main/java/com/clicktravel/cheddar/server/rest/RestServer.java

```diff
@@ -67,6 +73,39 @@ public void start(final int servicePort, final int statusPort, final String bind
         httpServer.start();
     }
 
+    private void enableAutoGenerationOfSwaggerSpecification() {
+        // The main scanner class used to scan the classes for swagger + jax-rs annoatations
+        final io.swagger.jaxrs.config.BeanConfig beanConfig = new BeanConfig();
+        beanConfig.setResourcePackage("com.clicktravel.services,com.clicktravel.services.*");
+        beanConfig.setSchemes(new String[] { "https" });
+        beanConfig.setBasePath("/");
+        final Info info = new Info();
+        info.setVersion("1.0.0");
+        beanConfig.setInfo(info);
+        beanConfig.setTitle("Swagger Specification");
+        beanConfig.setVersion("1.0.0");
+
+        // The follow sets up the security schemes so they can be referenced later (doesn't affect codegen methods!)
+        final Swagger swaggerConfiguration = beanConfig.getSwagger();
+        final OAuth2Definition oauth2ImplicitDefinition = new OAuth2Definition();
+        final String applicationGatewayEndpiont = "https://auth.travel.cloud";
```

Comment:
NOTE: This Cheddar class implementation is strongly coupled with our infrastructural choices.